core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -93,6 +93,7 @@
     private long requestTimeoutMillis = DEFAULT_REQUEST_TIMEOUT;
     private long sendMessageTimeoutMillis = DEFAULT_SEND_MESSAGE_TIMEOUT;
     private boolean tlsEnabled = false;
+    private String sniServerName;
```

Comment:
having taken a look at the JavaDoc of `io.vertx.proton.ProtonClientOptions`, I believe that we should set the *virtualHost* property instead of the *sniServerName* property. My understanding is that this will set both the server name in the SNI extension at the TLS level as well as the *hostname* property in the AMQP open frame. The latter is much easier to access from application level code (e.g. the protocol adapters) in order to e.g. determine the tenant from the hostname in the future ...

Reply:
In TLS SNI communication we have 2 scenarios : 

1.  Client establish TLS SNI connection and  `virtualHost ` are equal with  `sniServerName `
In this case ,  we need only to set  `virtualHost ` that automatically set  `sniServerName `

2. Client establish TLS SNI connection and  `virtualHost `are different from   `sniServerName ` 
In this case , we need to set `virtualHost ` that set also  `sniServerName ` and after override `sniServerName `  by setting this via `setSniServerName(...)`

Reply:
The second case is not a supported scenario in AMQP. In fact, its semantics is not specified.
Fro the AMQP 1.0 spec, section 2.7.1 (*open* frame):
> hostname: the name of the target host
The name of the host (either fully qualified or relative) to which the sending peer is connecting. It is
not mandatory to provide the hostname. If no hostname is provided the receiving peer SHOULD
select a default based on its own configuration. This field can be used by AMQP proxies to
determine the correct back-end service to connect the client to.
This field MAY already have been specified by the sasl-init frame, if a SASL layer is used, or,
the server name indication extension as described in RFC-4366, if a TLS layer is used, in which
case this field SHOULD be null or contain the same value. It is undefined what a different value
to that already specified means.

I therefore do not think that we should allow clients to set an SNI server name that is different from the virtual host. Would you agree?

Reply:
Yes, for AMQP 1.0 totally make sense.

Reply:
In case that TLS SNI will be mainly use by protocols adapter maybe make sens to move the configuration from `ServerConfig 
`  to `ProtocolAdapterProperties` ?

Reply:
But that would also limit it to be used by protocol adapters as well. FMPOV SNI is a concept of general interest on the server side. Thus, I would rather keep it in `ServerConfig`.